RCP_2024_NATIONAL_updated.py

```diff
@@ -0,0 +1,95 @@
+import requests
+import re
+from bs4 import BeautifulSoup #data scraper package
+import numpy
+import pandas as pd
+from itertools import zip_longest
+#url loading and get
+url = 'https://www.realclearpolling.com/polls/president/general/2024/trump-vs-biden'
+
+def get_national2024_data(url):
+    import json
+    page = requests.get(url)
+
+    #check and return code
+    if page.status_code == 200:
+        url_content = page.text
+        print("Success")
+    else:
+        print("Failed to retrieve. Status Code was", page.status_code)
+        exit()
+
+    soup = BeautifulSoup(page.content, "html.parser") #EVERYTHING
+
+    script_tags = soup.find_all('script')
+
+
+    for script in script_tags:
+        if script.string and 'finalData' in script.string:
+            
+            str = script.string
+            break
+
+    str2 = str.split('self.__next_f.push(')
+    str3 = str2[1][:-1]
+    json = json.loads(str3)
+
+    json_str = json[1] 
+    print(json_str)
+
+    #search pattern index
+    pollster_pattern = r'"pollster":\s*"([^"]*)"'
+    date_pattern = r'"date":\s*"([^"]*)"'
+    sample_size_pattern = r'"sampleSize":\s*"([^"]*)"'
+    margin_error_pattern = r'"marginError":\s*"([^"]*)"'
+    dvalue_pattern = r'"candidate":\[{"name":"Biden","affiliation":"Democrat","value":"([^"]*)"'
+    rvalue_pattern = r'"candidate":\[{[^}]*},{"name":"Trump","affiliation":"Republican","value":"([^"]*)"'
```

Comment:
you can remove these -- you create new patterns later in the code that you actually use.

Reply:
thanks boss